python/lsst/pipe/base/graph/graph.py

```diff
@@ -0,0 +1,779 @@
+            # add a key to the defaultdict that is a `DatasetRef` for one of
+            # the individual datasets inside the `Quantum`, with a value of a
+            # `_DatasetTracker`. Depending on the type of the individual dataset
+            # (it's connection type) add a `QuantumNode` for this quantum to the
+            # `_DataSetTracker`s input or output.
+            for quantum in quantumSet:
+                inits = quantum.initInputs.values()
+                predicted = quantum.predictedInputs.values()
+                value = QuantumNode(quantum, quantumCounter)
+                self._nodeList.append(value)
+
+                for dsRef in chain(inits, predicted):
+                    # unfortunately, `Quantum` allows inits to be individual
+                    # `DatasetRef`s or an Iterable of such, so there must
+                    # be an instance check here
+                    if isinstance(dsRef, Iterable):
+                        for sub in dsRef:
+                            self._datasetRefDict[sub].inputs.add(value)
+                    else:
+                        self._datasetRefDict[dsRef].inputs.add(value)
+                for dsRef in chain.from_iterable(quantum.outputs.values()):
+                    self._datasetRefDict[dsRef].output.add(value)
+                self._quantumToTask[quantumCounter] = taskDef
+                quantumCounter += 1
+
+        # Graph of task relations, used in various methods
+        self._taskGraph = nx.DiGraph()
+        # _datasetDictToEdgeIterator takes the stored inputs and outputs
+        # associated with each key in the default dict to yield a tuple which
+        # represents an edge between each yielded node
+        self._taskGraph.add_edges_from(_datasetDictToEdgeIterator(self._datasetDict))
+        # Attempting to create a graph that contains a single task will create an
+        # edge from None to the task, but None is not a node we actually want
+        # so remove it here. This is a consequence of using add_edges from and
+        # is an implementation detail.
+        if None in self._taskGraph.nodes():
+            self._taskGraph.remove_node(None)
+
+        # Graph of quanta relations
+        self._connectedQuanta = nx.DiGraph()
+        # Likewise to the `_taskGraph` above
+        self._connectedQuanta.add_edges_from(_datasetDictToEdgeIterator(self._datasetRefDict))
+        # Attempting to create a graph that contains a quantum unconnected from
+        # any other node will create an edge from None to the task, but None is
+        # not a node we actually want so remove it here. This is a consequence
+        # of using add_edges from and is an implementation detail.
+        if None in self._connectedQuanta.nodes():
+            self._connectedQuanta.remove_node(None)
+
+    @property
+    def taskGraph(self) -> nx.DiGraph:
+        """Returns a graph representing the relations between the tasks inside
```

Comment:
Because QuantumGraph is a extension/abstraction around the `graph` property (if you iterate over both containers you get the relations between `QantumNodes` so in some cases they could be interchangeable. The `taskGraph` property is just a convenience that exposes some information, and could never be used as or confused with `QuantumGraph`